zapier/cradlApi.js

```diff
@@ -0,0 +1,78 @@
+const http = require('https'); // require('http') if your URL is not https
+
+const downloadFile = async (url, z) => {
+  const getResponse = await z.request({
+    url: url,
+    raw: true,
+  })
+  return getResponse
+}
+
+async function makePostRequest(z, endpoint, body) {
+  return z.request({
+    url: process.env.API_BASE_URL + endpoint,
+    method: 'POST',
+    body: body,
+  });
+}
+
+async function makeGetRequest(z, endpoint) {
+  return z.request({
+    url: process.env.API_BASE_URL + endpoint,
+    method: 'GET',
+  });
+}
+
+async function putToFileServer(z, url, content) {
+  return z.request({
+    url: url,
+    method: 'PUT',
+    body: content,
+    raw: true,
+  });
+}
+
+async function createDocument(z, inputFileUrl) {
+  const postDocumentsResponse = await makePostRequest(z, '/documents', {})
+  // bundle.inputData.file will be a URL where the file data can be downloaded
+  fileResponse = await downloadFile(inputFileUrl, z)
+  const fileServerResponse = await putToFileServer(z, postDocumentsResponse.json.fileUrl, fileResponse.buffer())
+  return postDocumentsResponse.json.documentId
+}
+
+async function createPrediction(z, documentId, modelId, trainingId){
+  return makePostRequest(z, '/predictions', {
+    documentId: documentId,
+    modelId: modelId,
+    trainingId: trainingId,
+  });
+}
+
+async function executeWorkflow(z, documentId, workflowId){
```

Comment:
Don't we want to add keyword-arguments here that is passed to the workflow along with the document? Or is that an advanced feature for later?

Reply:
Do you have any examples of kwargs? We could have a curated list of them, but I don't think free-form kwargs will be looked kindly upon in Zapier's app review when publishing.